quinn/src/endpoint.rs

```diff
@@ -437,16 +473,17 @@ impl State {
             std::array::from_fn(|_| bufs.next().expect("BATCH_SIZE elements"))
         };
         loop {
-            match self.socket.poll_recv(cx, &mut iovs, &mut metas) {
+            match socket.poll_recv(rcx.cx, &mut iovs, rcx.metas) {
                 Poll::Ready(Ok(msgs)) => {
                     self.recv_limiter.record_work(msgs);
-                    for (meta, buf) in metas.iter().zip(iovs.iter()).take(msgs) {
+                    for (meta, buf) in rcx.metas.iter().zip(iovs.iter()).take(msgs) {
                         let mut data: BytesMut = buf[0..meta.len].into();
                         while !data.is_empty() {
+                            received_data = true;
```

Comment:
This will trigger on any UDP datagram, including random garbage that isn't from the remote server or even a valid QUIC packet. Would it make more sense to rename it `received_connection_packet` or similar, and only set it in the `ConnectionEvent` case?